src/expr/src/expr/expr_array.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::TryFrom;
+use std::sync::Arc;
+
+use risingwave_common::array::column::Column;
+use risingwave_common::array::{
+    ArrayBuilder, ArrayImpl, ArrayMeta, ArrayRef, DataChunk, ListArrayBuilder,
+};
+use risingwave_common::error::{Result, RwError};
+use risingwave_common::types::DataType;
+use risingwave_common::{ensure, try_match_expand};
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost as expr_build_from_prost, BoxedExpression, Expression};
+
+#[derive(Debug)]
+pub struct ArrayExpression {
+    element_type: Box<DataType>,
+    elements: Vec<BoxedExpression>,
+}
+
+impl Expression for ArrayExpression {
```

Comment:
@TennyZhuang another expression not generated by macros

Reply:
Variadic expressions are not supported by the macro yet: case-when, coalesce, concat_ws, array.